OpenNeuroOrg/openneuro
# Extract mongodb  dependency and replace with mongoose

## Major Changes
- remove mongodb dependency
- delete libs/mongo.js
- remove mongo.connect from:
  - server.js
  - migrations/upload.js
- add mongoose models for:
  - Doi
  - File
  - Issue
  - Key
  - MailgunIdentifier
  - Notification
  - Summary
- replace all usages of `mongo.collections.crn` with mongoose equivalents, involving
  - analytics
  - dataset: getDatasets, checkDatasetRead, public update, followers, revisions
  - snapshot: gets, metadata
  - file: updates
  - issues: (general & snapshot)
  - permission: get, update, and remove
  - summary: get, update
  - validation: update
  - comment: create reply
  - doi: create for snapshot, get
  - sitemap
  - subscriptions: get, deletes
  - apikey: generate
  - notifications: snapshot creation, comment creation, dataset deletion, owner unsubscription
  - auth states: dataset access, comment access

## Minor Changes
- replace `mongo.ObjectID` with `mongoose.Types.ObjectID`

##Testing
- checked for regression in
  - login/logout
  - create-user
  - public/datasets (with and without login)
  - dashboard/datasets
  - datasets/ds*
  - upload dataset
    - successfully uploaded
    - files present and accessibly
    - expected bids warning present
    - doi present
    - summary present
  - update dataset
    - add/remove authors
    - upload/delete README.md
  - publish dataset
  - create dataset snapshot
  - add comment & reply
  - save dataset
